src/inmanta/protocol/methods_v2.py

```diff
@@ -1365,20 +1368,20 @@ def update_notification(
 
 
 @typedmethod(
-    path="/code/<version>",
-    operation="GET",
-    agent_server=True,
+    path="/modulecodebatched/",
+    operation="PUT",
     arg_options=methods.ENV_OPTS,
-    client_types=[ClientType.agent],
+    client_types=[ClientType.compiler],
     api_version=2,
+    varkw=True,
 )
-def get_source_code(tid: uuid.UUID, version: int, resource_type: str) -> list[model.Source]:
+def upload_modules(tid: uuid.UUID, **kwargs: object) -> ReturnValue[int]:
```

Comment:
you need pydantic objects/dataclasses for all the stuff going in

but, more important, why does this method exist at all? 
i.e. unlike the files, we always push all this data (we don't first check which modules don't exists)
so if we make this part of the put_version directly the traffic pattern will be the same, except that we make one less call.  What is the rationale behind this choice?

Reply:
I see, moved this to be part of `put_version`.